### PR TITLE
Config refactoring

### PR DESCRIPTION
**Pull request by FalseR20**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,12 +37,11 @@ sudo -u postgres psql
 create database sc_auth;
 create user sc_auth with encrypted password 'sc_auth';
 grant all privileges on database sc_auth to sc_auth;
-\q
 ```
 
 **Activation**
 
-Set postgres in `sc_server_auth/config.py:24:28`  <!--- temporally, config in progress -->
+Set postgres in sc_server_auth/configs/settings.toml or as flag in server run script
```
Comment (AlexandrZagorskiy) [inline README.md]:
```suggestion
Set postgres in sc_server_auth/configs/settings.toml or as argument in server run script
```

File: sc_server_auth/__main__.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+from sc_server_auth.configs.models import Database, RunArgs
+from sc_server_auth.configs.parser import Parser, get_config
+
+
+def main():
+    config = get_config()
+    args = RunArgs()
+
+    parser = ArgumentParser(prog="python3 -m sc_server_auth", description="Run SC Server")
```
Comment (AlexandrZagorskiy) [inline sc_server_auth/__main__.py]: It is not a SC server, rename to the auth-server applicatin

File: sc_server_auth/configs/constants.py [hunk 3]
```diff
@@ -1,42 +1,42 @@
...
+TOKENS = "tokens"
+SERVER = "server"
+SC_SERVER = "sc_server"
 DATABASE = "database"
+
+# common
+LOG_LEVEL = "log_level"
+
+# database
+POSTGRES = "postgres"
```
Comment (AlexandrZagorskiy) [inline sc_server_auth/configs/constants.py:13]: Should we add sqlite here? I think we should

File: sc_server_auth/configs/models.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+@dataclass
+class ServerParams:
+    protocol: str
+    host: str
+    port: int
+
+
+class Database(Enum):
+    SQLITE = "sqlite"
+    POSTGRES = "postgres"
```
Comment (AlexandrZagorskiy) [inline sc_server_auth/configs/models.py]: should we read constants from the _constants.py_ module?

File: sc_server_auth/configs/paths.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ROOT_DIR_PATH = CONFIG_DIR_PATH.parent
+
+PRIVATE_KEY_PATH = ROOT_DIR_PATH.joinpath("private.pem")
+PUBLIC_KEY_PATH = ROOT_DIR_PATH.joinpath("public.pem")
+
+LOG_DIR_PATH = ROOT_DIR_PATH.joinpath("loggings")
+LOG_DIR_PATH.mkdir(parents=True, exist_ok=True)
+LOG_PATH = LOG_DIR_PATH.joinpath("external-modules.log")
+
+SQLITE_PATH = ROOT_DIR_PATH.parent.joinpath("database.db")
```
Comment (AlexandrZagorskiy) [inline sc_server_auth/configs/paths.py]: remove parent, I think we can face with some problems with docker if database will be stored outside of root_path

File: sc_server_auth/configs/settings.toml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# DEV
+
+[common]
+log_level = "INFO"
+
+[tokens]
+access_token_life_span = 1800
+refresh_token_life_span = 2_592_000
+bits = 2048
```
Comment (AlexandrZagorskiy) [inline sc_server_auth/configs/settings.toml:9]: What should happened if we will change a bits value? Should we configure this value?

File: sc_server_auth/server/auth.py [hunk 7]
```diff
@@ -64,36 +64,32 @@ async def get_tokens(credentials: models.CredentialsModel):
...
             public_key = file.read()
     except FileNotFoundError:
         log.error(Exception(FileNotFoundError))
         raise Exception(FileNotFoundError)
     request_params = token.dict()
     log.debug(f"GetAccessToken request: " + str(request_params))
-    username = jwt.decode(request_params[cnt.TOKEN], public_key, issuer=params[cnt.ISSUER], algorithm="RS256")[
-        cnt.USERNAME
-    ]
+    username = jwt.decode(request_params[cnt.TOKEN], public_key, issuer=config.issuer, algorithm="RS256")[cnt.USERNAME]
```
Comment (AlexandrZagorskiy) [inline sc_server_auth/server/auth.py]: Should we extract algorithm in config? If not, at least you can import algo as constant

File: sc_server_auth/server/models.py [hunk 8]
```diff
@@ -5,23 +5,26 @@
...
+        with open(PUBLIC_KEY_PATH, "rb") as file:
             public_key = file.read()
-            jwt.decode(value, public_key, issuer=params[cnt.ISSUER], algorithm="RS256")
+            jwt.decode(value, public_key, issuer=config.issuer, algorithm="RS256")
     except (
         jwt.exceptions.InvalidTokenError,
         jwt.exceptions.InvalidSignatureError,
         jwt.exceptions.InvalidIssuerError,
         jwt.exceptions.ExpiredSignatureError,
         FileNotFoundError,
```
Comment (AlexandrZagorskiy) [inline sc_server_auth/server/models.py]: Should auth-service start if file with key wasn't found? Is it good idea to check an existence of the public key before starting service?

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (FalseR20) [reply README.md]: Done

File: sc_server_auth/__main__.py (hunk 2 above)
Comment (FalseR20) [reply sc_server_auth/__main__.py]: Renamed

File: sc_server_auth/configs/constants.py (hunk 3 above)
Comment (FalseR20) [reply sc_server_auth/configs/constants.py:13]: Why not

File: sc_server_auth/configs/models.py (hunk 4 above)
Comment (FalseR20) [reply sc_server_auth/configs/models.py]: Done

File: sc_server_auth/configs/paths.py (hunk 5 above)
Comment (FalseR20) [reply sc_server_auth/configs/paths.py]: I did it and moved database

File: sc_server_auth/configs/settings.toml (hunk 6 above)
Comment (FalseR20) [reply sc_server_auth/configs/settings.toml:9]: It's parameter in key genering:
```python
pkey.generate_key(type=crypto.TYPE_RSA, bits=config.bits)
```

File: sc_server_auth/server/auth.py (hunk 7 above)
Comment (FalseR20) [reply sc_server_auth/server/auth.py]: I made it as constant

File: sc_server_auth/server/models.py (hunk 8 above)
Comment (FalseR20) [reply sc_server_auth/server/models.py]: Yes, good idea

**Review by AlexandrZagorskiy — APPROVED** [review]

**Pull request merged**